src/components/icons.js

```diff
@@ -36,15 +36,19 @@ export const containsUnlabelledIcon = ({ children, 'aria-label': ariaLabel, 'ari
   return false;
 };
 
+/**
+ * @typedef IconOptions
+ * @property {string} [className] The class of the icon
+ * @property {number} [size] The size of the icon
+ * @property {object} [style] The icon style
+ * @property {string} [aria-label] An optional accessible label to apply to the icon.
+ *    If not specified 'aria-hidden' will be set to true.
+ */
+
 /**
  * Creates an icon: FA or custom.
  * @param {string} shape - see {@link https://fontawesome.com/icons?d=gallery}
- * @param {object} [props]
- * @param {string} [props.className] The class of the icon
- * @param {number} [props.size] The size of the icon
- * @param {object} [props.style] The icon style
- * @param {string} [props.aria-label] An optional accessible label to apply to the icon.
- *    If not specified 'aria-hidden' will be set to true.
+ * @param {IconOptions} [props]
```

Comment:
TS was getting confused by the - in props.aria-label and thinking there was a prop named aria. Changing the JSDoc to use a typedef for props instead of inlining the object fields fixed it.